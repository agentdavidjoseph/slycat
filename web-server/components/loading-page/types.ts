/**
 * @export
 * @interface JobCodesProps
 */
export interface JobCodesProps {}

/**
 *
 * @export
 * @interface logMap
 */
export interface logMap {
  logLineArray: string[];
}

export interface LogListProps {
  log: logMap;
  jobStatus: string;
  sessionExists: boolean;
}

export interface InfoBarProps {
  jid: string;
  hostname: string;
  sessionExists: boolean;
}
