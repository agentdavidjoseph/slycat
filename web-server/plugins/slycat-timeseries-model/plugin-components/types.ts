
/**
 * @export
 * @interface LoadingPageProps
 */
export interface LoadingPageProps {
  jid: string
  hostname: string
  modelId: string
}
/**
 *
 * @export
 * @interface LoadingPageState
 */
export interface logMap {
  pending: [string]
  failed: [string]
  running: [string]
  complete: [string]
}
/**
 *
 * @export
 * @interface LoadingPageState
 */
export interface LoadingPageState {
  log: logMap
  sessionExists: boolean
  progressBarHidden: boolean
  progressBarProgress: number
  modalId: string
  modelId: string
  modelShow: boolean
}