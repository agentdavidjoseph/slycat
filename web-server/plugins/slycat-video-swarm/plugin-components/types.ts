
/**
 * @export
 * @interface LoadingPageProps
 */
export interface LoadingPageProps {
  jid: string;
  hostname: string;
  modelId: string;
  modelState: string;
  callback?: VoidFunction;
}
/**
 *
 * @export
 * @interface logMap
 */
export interface logMap {
  logLineArray: [string];
}
/**
 *
 * @export
 * @interface LoadingPageState
 */
export interface LoadingPageState {
  log: logMap;
  sessionExists: boolean;
  progressBarHidden: boolean;
  progressBarProgress: number;
  modalId: string;
  modelMessage: string;
  modelId: string;
  modelShow: boolean;
  jobStatus: string;
  pullCalled: number;
  modelState: string;
}

export interface LogListProps {
  log: logMap;
  jobStatus: string;
  sessionExists: boolean;
}

export interface LoadButtonProps {
  disabled: boolean;
  onClick: () => void;
}

export interface LoadingPageButtonsProps {
  modalId: string
  jobStatus: string
  cancelJob: () => void
}
