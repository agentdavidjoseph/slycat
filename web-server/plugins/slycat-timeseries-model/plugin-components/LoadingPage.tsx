'use strict';
import * as React from 'react';
import client from '../../../js/slycat-web-client';
import ProgressBar from 'components/ProgressBar.tsx';

/**
 * @export
 * @interface LoadingPageProps
 */
export interface LoadingPageProps {
  jid: string
  hostname: string
}

/**
 *
 * @export
 * @interface LoadingPageState
 */
export interface LoadingPageState {
  sessionExists: boolean
  progressBarHidden: boolean
  progressBarProgress: number
}

/**
 * react component used to create a loading page
 *
 * @export
 * @class LoadingPage
 * @extends {React.Component<LoadingPageProps, LoadingPageState>}
 */
export default class LoadingPage extends React.Component<LoadingPageProps, LoadingPageState> {
  public constructor(props:LoadingPageProps) {
    super(props)
    this.state = {
      sessionExists: false,
      progressBarHidden: false,
      progressBarProgress: 0
    }
  }
  /**
   * method runs after the component output has been rendered to the DOM
   */
  componentDidMount() {
    this.checkRemoteStatus(this.props.hostname);
  }

  // tear down
  componentWillUnmount() {

  }

  /**
   * function used to test if we have an ssh connection to the hostname
   * @param hostname name of the host we want to connect to
   * @async
   * @memberof SlycatRemoteControls
   */
  private checkRemoteStatus = async (hostname:string) => {
    return client.get_remotes_fetch(hostname)
      .then((json:any) => {
        this.setState({
          sessionExists:json.status,
          progressBarProgress: 10
        }, () => {
          console.log('called after set state');
        });
    });
  };

  public render() {
    console.log(this.state.sessionExists);
    return (
      <div className="slycat-job-checker bootstrap-styles">
        <ProgressBar
          hidden={this.state.progressBarHidden}
          progress={this.state.progressBarProgress}
        />
        <div className="slycat-job-checker-output text-white bg-secondary" >
          <p>
          Job id <b>{this.props.jid} </b> <br/>
          Default host name <b>{this.props.hostname}</b><br/>
          session status is <b>{this.state.sessionExists?'true':'false'}</b>
          </p>
        </div>
      </div>
    );
  }
}
