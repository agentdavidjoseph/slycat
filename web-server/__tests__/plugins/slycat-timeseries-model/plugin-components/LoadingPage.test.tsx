
import React from 'react';
import LoadingPage  from "slycat-timeseries-model/plugin-components/LoadingPage.tsx";
import {LoadingPageProps, LoadingPageState}  from "slycat-timeseries-model/plugin-components/types.ts";
import { mount, shallow } from "enzyme";

describe('Timeseries LoadingPage',() =>{ 
  test('should be Defined', () => {
    expect(LoadingPage).toBeDefined();
  });
  //TODO: fix these tests, there is an error with with how jest now handles ajax calls
  xtest('we have expected props on initial load', () => {
    const render = mount(
      <LoadingPage
      modelId={'modelID'}
      modelState={'waiting'}
      jid={'1234'}
      hostname={'hostname'}
      />
    );
    const props = render.props() as LoadingPageProps;
    expect(props).toMatchSnapshot();
  });
  xtest('we have expected state  on initial load', () => {
    const render = mount(
      <LoadingPage
      modelId={'modelID'}
      modelState={'waiting'}
      jid={'1234'}
      hostname={'hostname'}
      />
    );
    const state = render.state() as LoadingPageState;
    expect(state).toMatchSnapshot();
  });
  xtest('we have expected props on initial load', () => {
    const myDate = new Date('2019-05-14T11:01:58/135Z');
    global.Date = jest.fn(() => myDate);
    const test = shallow(
      <LoadingPage
      modelId={'modelID'}
      modelState={'waiting'}
      jid={'1234'}
      hostname={'hostname'}
      />
    );
    expect(test).toMatchSnapshot();
  });
});
