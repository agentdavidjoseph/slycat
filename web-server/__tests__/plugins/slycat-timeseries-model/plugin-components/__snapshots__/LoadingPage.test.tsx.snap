// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeseries LoadingPage we have expected props on initial load 1`] = `
Object {
  "hostname": "hostname",
  "jid": "1234",
  "modelId": "modelID",
  "modelState": "waiting",
}
`;

exports[`Timeseries LoadingPage we have expected props on initial load 2`] = `
<div
  className="slycat-job-checker bootstrap-styles"
>
  <div
    className="slycat-job-checker-controls"
  >
    <InfoBar
      hostname="hostname"
      jid="1234"
      sessionExists={false}
    />
    <div
      style={
        Object {
          "paddingBottom": "15px",
          "paddingTop": "15px",
        }
      }
    >
      <ProgressBar
        hidden={false}
        progress={0}
      />
    </div>
    <div
      className="row justify-content-center"
    >
      <div
        className="btn-group col-8"
        role="group"
      >
        <LoadingPageButtons
          cancelJob={[Function]}
          connectModalCallBack={[Function]}
          hostname="hostname"
          jobStatus="Job Status Unknown"
          modalId="ConnectModal"
          modelShow={false}
          pullHPCData={[Function]}
          sessionExists={false}
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="collapse col-12"
        id="collapseJobCodes"
      >
        <div
          className="card card-body"
        >
          <JobCodes />
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-lg-12"
  >
    <div
      className="slycat-job-checker-output text-white bg-secondary"
    >
      <LogList
        jobStatus="Job Status Unknown"
        log={
          Object {
            "logLineArray": Array [],
          }
        }
        modalId="ConnectModal"
        modelId="modelID"
        modelMessage=""
        modelShow={false}
        modelState="waiting"
        progressBarHidden={false}
        progressBarProgress={0}
        pullCalled={0}
        sessionExists={false}
      />
    </div>
  </div>
</div>
`;

exports[`Timeseries LoadingPage we have expected state  on initial load 1`] = `
Object {
  "jobStatus": "Job Status Unknown",
  "log": Object {
    "logLineArray": Array [],
  },
  "modalId": "ConnectModal",
  "modelId": "modelID",
  "modelMessage": "",
  "modelShow": false,
  "modelState": "waiting",
  "progressBarHidden": false,
  "progressBarProgress": 0,
  "pullCalled": 0,
  "sessionExists": false,
}
`;
